src/Abstractions/TaskOptions.cs

```diff
@@ -66,6 +66,7 @@ public record SubOrchestrationOptions : TaskOptions
     public SubOrchestrationOptions(TaskRetryOptions? retry = null, string? instanceId = null)
         : base(retry)
     {
+        Check.NotEntity(instanceId);
```

Comment:
Right. Note that it does not break just because of this check. The backend also treats '@' instance ids differently. So the check is actually necessary for giving good error messages as opposed to mysterious internal errors.

Reply:
I suppose we could - would be a good use of AppContext. Default is opted in, add way to opt out.